### PR TITLE
Refactor of BLoC & Sensing | Fix authentication flow | Polar disconnection exception handling

### PR DESCRIPTION
**Pull request by bardram**
Major refactor of entire app - cleaning up a lot of things, especially everything that has to do with CAMS and sensing, incl. the BloC, Sensing, and Backend classes. Addresses issues;

* #215  & #205 & #202 
* #214 
* #213 (but not entirely fixed - waiting for a longer refresh token from @xelahalo )
* #212 

The core CAMS framework is also updated, which should address #201

### CODE REVIEW

**Comment by LarsRefsgaard** [comment]
@bardram Please do `git merge origin/master` and resolve the conflicts :)

**Comment by LarsRefsgaard** [comment]
@bardram Also the title of the PR needs to be updated to reflect what is happening more precisely

**Comment by bardram** [comment]
> @bardram Also the title of the PR needs to be updated to reflect what is happening more precisely

👍

File: pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: carp_study_app
 description: The generic CARP study app.
 publish_to: 'none'
-version: 1.2.0
+version: 1.4.0
```
Comment (LarsRefsgaard) [inline pubspec.yaml:4]: If you change the version, please also change the changelog and the iOS version in XCode. I've updated the version in XCode.

**Review by LarsRefsgaard — CHANGES_REQUESTED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (bardram) [reply pubspec.yaml:4]: Should be done now. Also updated the README a bit.

**Review by LarsRefsgaard — APPROVED** [review]

**Pull request merged**